src/ui/render.ts

```diff
@@ -278,20 +293,31 @@ export default class UI {
 
   private registerEvents() {
     this.lightSheet.events.on(EventType.CORE_SET_CELL, (event) => {
-      if (this.lightSheet.isReady) this.onCoreSetCell(event);
+      this.onCoreSetCell(event);
     });
   }
 
   private onCoreSetCell(event: LightsheetEvent) {
     const payload = event.payload as CoreSetCellPayload;
-    // Get HTML elements and (new) IDs for the payload's cell and row.
-    const elInfo = LightSheetHelper.getElementInfoForSetCell(payload);
+    // Create new columns if the column index is greater than the current column count.
+    const columnCount = this.getColumnCount();
+    if (payload.indexPosition.column > columnCount - 1) {
+      const diff = payload.indexPosition.column + 1 - columnCount;
+      for (let i = 0; i < diff; i++) {
+        this.addColumn();
+      }
+    }
 
-    if (!elInfo.rowDom) {
-      const row = this.addRow(payload.indexPosition.row);
-      elInfo.rowDom = row;
-      row.id = elInfo.rowDomId;
+    const rowCount = this.getRowCount();
+    if (payload.indexPosition.row > rowCount - 1) {
+      const diff = payload.indexPosition.row + 1 - rowCount;
+      for (let i = 0; i < diff; i++) {
+        this.addRow();
+      }
     }
+
+    // Get HTML elements and (new) IDs for the payload's cell and row.
+    const elInfo = this.getElementInfoForSetCell(payload);
     if (!elInfo.cellDom) {
```

Comment:
Won't `cellDom` always exist at this point since `addCell` is called by both `addRow` and `addColumn`?

Reply:
The cells will in-fact already exist as you said, but those cells created in `addRow` and `addColumn` are just mock empty cells to ensure the table looks correct, in case they are not being filled in later (for example, in case someone insert one cell in a row).

This second call, here, is setting the actual values and Ids and potential other things.

Reply:
True, I though something else is being done (some of the old code).

Deleted unnecessary check.